### ISSUE TITLE
ConnectionPatch across axes needs to be excluded from layout management

### ISSUE THREAD

**Issue by canismarko** [matplotlib/matplotlib#14907]
### Bug report

**Bug summary**

When adding a ``ConnectionPatch`` that spans multiple axes with ``constrained_layout=True``, the layout becomes broken, with the axes getting shrunk considerably. This can be resolved by calling ``patch.set_in_layout(False)`` before adding the artist to the axes. It may be useful to include a note about this to the [documentation for ``ConnectionPatch``](https://matplotlib.org/3.1.1/api/_as_gen/matplotlib.patches.ConnectionPatch.html).

Thanks to ImportanceOfBeingErnest for helping me figure this out on stackoverflow: [https://stackoverflow.com/questions/57237711/connectionpatch-breaks-constrained-layout-when-crossing-axes/57241978#57241978](https://stackoverflow.com/questions/57237711/connectionpatch-breaks-constrained-layout-when-crossing-axes/57241978#57241978)

**Code for reproduction**

I followed the example provided [here](https://matplotlib.org/3.1.1/gallery/userdemo/connect_simple01.html#sphx-glr-gallery-userdemo-connect-simple01-py), except that I changed line 4 to use a managed layout.

```python
fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(6, 3), constrained_layout=True)
```

**Actual outcome**

Running the code with constrained_layout produced the following figure:
![image](https://user-images.githubusercontent.com/2363800/62009099-e0cb4b00-b120-11e9-88e7-54ebdd7304ea.png)

However, if I add ``con.set_in_layout(False)`` before the ``ax2.add_artist(con)`` on line 26, it produces a nicely layed out figure:

![image](https://user-images.githubusercontent.com/2363800/62009125-356ec600-b121-11e9-82a3-9a6e7b231bd9.png)

**Expected outcome**

This is the expected figure with ``constrained_layout=False`` from the example:

![image](https://user-images.githubusercontent.com/2363800/62009132-533c2b00-b121-11e9-8143-a6689c82a028.png)

**Matplotlib version**

  * Operating system: Linux, Gentoo 2.6
  * Matplotlib version: 3.1.1
  * Matplotlib backend: module://ipykernel.pylab.backend_inline
  * Python version: 3.6.7
  * Jupyter version (if applicable): 5.2.3
  * Other libraries: 

Packages were installed in a conda environment from conda-forge channel.

**Comment by jklymak** [comment]
Want to try a PR along those lines?  Might also be worth looking at constrained layout tutorial and letting us know where this could be better explained.  I think another solution to your issue would have been if you had made the connector a figure artist rather than an axes artist. Then the layout manager wouldn’t have considered the patch as something it needed to make room for.

**Comment by canismarko** [comment]
Yeah that sounds great. It'll be my first PR to a major project so feedback would be much appreciated once I get the PR done..

**Comment by jklymak** [comment]
Yeah, its a bit of a process, and please ping us (and ping us) if it doesn't get attention.

**Comment by canismarko** [comment]
I created a PR that addresses this in the documentation: https://github.com/matplotlib/matplotlib/pull/14957

If I don't see any movement in the next few days I'll ping the mailing list?

**Comment by jklymak** [comment]
You can ping in the pr

**Comment by ImportanceOfBeingErnest** [comment]
Closing as mainly fixed by #14957
